chain/network/src/peer_manager/connection/mod.rs

```diff
@@ -257,8 +258,8 @@ impl Drop for OutboundHandshakePermit {
 #[derive(Clone)]
 pub(crate) struct Pool(Arc<ArcMutex<PoolSnapshot>>);
 
-#[derive(thiserror::Error, Clone, Debug, PartialEq, Eq)]
-pub(crate) enum PoolError {
+#[derive(thiserror::Error, Clone, Debug, PartialEq, Eq, BorshSerialize, BorshDeserialize)]
```

Comment:
please don't make it BorshSerializable. Every time you make something borshserializable and send it over the network, backward compatibility becomes a hell to maintain.